lib/beaker/host/mac/pkg.rb

```diff
@@ -10,6 +10,25 @@ def install_package(name, cmdline_args = '', version = nil)
     execute("installer -pkg /Volumes/#{name}/#{name}.pkg -target /")
   end
 
+  # Install a package from a specified dmg
+  #
+  # @param [String] dmg_file      The dmg file, including path if not
+  #                               relative. Can be a URL.
+  # @param [String] pkg_base      The base name of the directory that the dmg
+  #                               attaches to under `/Volumes`
+  # @param [String] pkg_name      The name of the package file that should be
+  #                               used by the installer
+  # @example: Install vagrant from URL
+  #   mymachost.generic_install_dmg('https://releases.hashicorp.com/vagrant/1.8.4/vagrant_1.8.4.dmg', 'Vagrant', 'Vagrant.pkg')
+  def generic_install_dmg(dmg_file, pkg_base, pkg_name)
```

Comment:
Hmm. l don't see `install_package` and `install_dmg` being equivalent.

The interface for `install_package` takes the name of a package to install, which I see as equivalent to using a package management system (e.g `apt-get install foo`, `yum install foo`). Whereas, `install_dmg` takes a file identifier, which I see as equivalent to using a lower level package manager (e.g `dpkg -i foo.deb`, `rpm -i foo.rpm`). Combining this functionality would require a considerable amount of work across the `install_package` implementations for all of host operating systems.

I will defer to your judgment, @kevpl , but this seems like a large can of worms to open.